pkg/operator/ceph/object/notification/controller_test.go

```diff
@@ -0,0 +1,502 @@
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "CephBucketNotification",
+		},
+		Spec: cephv1.BucketNotificationSpec{
+			Topic: topicName,
+		},
+	}
+	cephCluster := &cephv1.CephCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      namespace,
+			Namespace: namespace,
+		},
+		Status: cephv1.ClusterStatus{
+			Phase: k8sutil.ReadyStatus,
+			CephStatus: &cephv1.CephStatus{
+				Health: "HEALTH_OK",
+			},
+		},
+	}
+	bucketName := "my-bucket"
+	scName := "my-storage-class"
+	obc := &bktv1alpha1.ObjectBucketClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      bucketName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				notificationLabelPrefix + notificationName: notificationName,
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind: "ObjectBucketClaim",
+		},
+		Spec: bktv1alpha1.ObjectBucketClaimSpec{
+			StorageClassName:   scName,
+			GenerateBucketName: bucketName,
+		},
+		Status: bktv1alpha1.ObjectBucketClaimStatus{
+			Phase: bktv1alpha1.ObjectBucketClaimStatusPhasePending,
+		},
+	}
+	req := reconcile.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      notificationName,
+			Namespace: namespace,
+		},
+	}
+
+	s := scheme.Scheme
+	s.AddKnownTypes(
+		cephv1.SchemeGroupVersion,
+		&cephv1.CephBucketNotification{},
+		&cephv1.CephBucketNotificationList{},
+		&cephv1.CephBucketTopic{},
+		&cephv1.CephBucketTopicList{},
+		&cephv1.CephCluster{},
+		&cephv1.CephClusterList{},
+		&bktv1alpha1.ObjectBucketClaim{},
+		&bktv1alpha1.ObjectBucketClaimList{},
+		&bktv1alpha1.ObjectBucket{},
+		&bktv1alpha1.ObjectBucketList{},
+	)
+
+	t.Run("provision notification when OBC exists but no OB", func(t *testing.T) {
+		objects := []runtime.Object{
+			bucketNotification,
+			bucketTopic,
+			cephCluster,
+			obc,
+		}
+
+		c := &clusterd.Context{
+			Executor:      &exectest.MockExecutor{},
+			RookClientset: rookclient.NewSimpleClientset(),
+			Clientset:     test.New(t, 3),
+		}
+
+		secrets := map[string][]byte{
+			"fsid":         []byte("name"),
+			"mon-secret":   []byte("monsecret"),
+			"admin-secret": []byte("adminsecret"),
+		}
+		secret := &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "rook-ceph-mon",
+				Namespace: namespace,
+			},
+			Data: secrets,
+			Type: k8sutil.RookType,
+		}
+
+		_, err := c.Clientset.CoreV1().Secrets(namespace).Create(ctx, secret, metav1.CreateOptions{})
+		assert.NoError(t, err)
+
+		cl := fake.NewClientBuilder().WithScheme(s).WithRuntimeObjects(objects...).Build()
+
+		r := &ReconcileNotifications{client: cl, context: c, opManagerContext: ctx}
+
+		res, err := r.Reconcile(ctx, req)
+		assert.NoError(t, err)
+		assert.True(t, res.Requeue)
+		// notification configuration is set
+		err = r.client.Get(ctx, types.NamespacedName{Name: notificationName, Namespace: namespace}, bucketNotification)
+		assert.NoError(t, err, bucketNotification)
+	})
+
+	/*t.Run("provision notification when OB exists", func(t *testing.T) {
+		ob := bktv1alpha1.ObjectBucket{
```

Comment:
These "pointer receiver" errors can be a little confusing. What this is saying is that an `ObjectBucket` does not match the `DeepCopyObject()` method because it takes a pointer receiver, where the object is not itself a pointer.

`pkg/operator/ceph/object/notification/controller_test.go:460:4: cannot use ob (variable of type "github.com/kube-object-storage/lib-bucket-provisioner/pkg/apis/objectbucket.io/v1alpha1".ObjectBucket) as "k8s.io/apimachinery/pkg/runtime".Object value in array or slice literal: missing method DeepCopyObject (DeepCopyObject has pointer receiver)`

The fix is actually as simple as changing this line to `ob := &bktv1alpha1.ObjectBucket`